### DISCUSSION TITLE
How to correlate node representation and graph representation?

### DISCUSSION THREAD

**Discussion by Bowen-n** [pyg-team/pytorch_geometric#4577] [Q&A] [answered]
I have a GNN model with a global pooling method, so I can compute the node representations and the graph representations.
Now, for each graph in a batch, I want to update the node repr by correlating node repr and graph repr.

How can I do this for a batch of graphs?

**Comment by rusty1s** [comment, marked as answer]
You can consider concatenating both as part of your model:
```python
x_global = global_mean_pool(x, batch)
x = torch.cat([x, x_global[batch]], dim=-1)
```

**Reply by Bowen-n** [reply]
It solves my problem, thanks !!
